File: src/test/java/no/nav/syfo/pdl/PdlHentPersonTest.java
package no.nav.syfo.pdl;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

import static java.util.Collections.emptyList;
import static java.util.Collections.singletonList;
import static org.assertj.core.api.Java6Assertions.assertThat;

@RunWith(MockitoJUnitRunner.class)
public class PdlHentPersonTest {
    @Test
    public void full_name_without_middlename() {
        PdlHentPerson pdlHentPerson = mockPdlHentPerson(null);

        String name = pdlHentPerson.getName();

        assertThat(name.split(" ").length).isEqualTo(2);
        assertThat(name).isEqualTo("Fornavn Etternavn");
    }

    @Test
    public void full_name_with_middleName() {
        PdlHentPerson pdlHentPerson = mockPdlHentPerson("Mellomnavn");

        String name = pdlHentPerson.getName();

        assertThat(name.split(" ").length).isEqualTo(3);
        assertThat(name).isEqualTo("Fornavn Mellomnavn Etternavn");
    }

    @Test
    public void null_when_hentPerson_is_null() {
        PdlHentPerson pdlHentPerson = new PdlHentPerson().hentPerson(null);

        String name = pdlHentPerson.getName();

        assertThat(name).isNull();
    }

    @Test
    public void null_when_PdlPersonNavnList_is_null() {
        PdlHentPerson pdlHentPerson = new PdlHentPerson().hentPerson(new PdlPerson().navn(null));

        String name = pdlHentPerson.getName();

        assertThat(name).isNull();
    }

    @Test
    public void null_when_PdlPersonNavnList_is_empty() {
        PdlHentPerson pdlHentPerson = new PdlHentPerson().hentPerson(new PdlPerson().navn(emptyList()));

        String name = pdlHentPerson.getName();

        assertThat(name).isNull();
    }

    private PdlHentPerson mockPdlHentPerson(String middleName) {
        return new PdlHentPerson()
                .hentPerson(new PdlPerson()
                        .navn(singletonList(new PdlPersonNavn()
                                .fornavn("Fornavn")
                                .mellomnavn(middleName)
                                .etternavn("Etternavn"))
                        ));
    }
}

File: src/test/kotlin/no/nav/syfo/pdl/PdlHentPersonTest.kt
package no.nav.syfo.pdl

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class PdlHentPersonTest {
    @Test
    fun full_name_without_middlename() {
        val pdlHentPerson = mockPdlHentPerson(null)
        val name: String? = pdlHentPerson?.fullName()
        assertThat(name!!.split(" ").toTypedArray().size).isEqualTo(2)
        assertThat(name).isEqualTo("Fornavn Etternavn")
    }

    @Test
    fun full_name_with_middleName() {
        val pdlHentPerson = mockPdlHentPerson("Mellomnavn")
        val name: String? = pdlHentPerson?.fullName()
        assertThat(name!!.split(" ").toTypedArray().size).isEqualTo(3)
        assertThat(name).isEqualTo("Fornavn Mellomnavn Etternavn")
    }

    @Test
    fun null_when_hentPerson_is_null() {
        val pdlHentPerson: PdlHentPerson = PdlHentPerson(null)
        val name: String? = pdlHentPerson.fullName()
        assertThat(name).isNull()
    }

    @Test
    fun null_when_PdlPersonNavnList_is_empty() {
        val pdlHentPerson: PdlHentPerson = PdlHentPerson(PdlPerson(emptyList(), null))
        val name: String? = pdlHentPerson.fullName()
        assertThat(name).isNull()
    }

    private fun mockPdlHentPerson(middleName: String?): PdlHentPerson? {
        return PdlHentPerson(PdlPerson(
                listOf(PdlPersonNavn(
                        fornavn = "Fornavn",
                        mellomnavn = middleName,
                        etternavn = "Etternavn"
                )),
                null
        ))
    }
}
